### PR TITLE
feat(nav): navigation component

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #6531 

## Summary
WIP

### CODE REVIEW

File: src/components/nav-menu-item/nav-menu-item.tsx [hunk 1]
```diff
@@ -13,7 +13,12 @@ import {
   VNode
 } from "@stencil/core";
 import { FlipContext } from "../interfaces";
-import { getElementDir, getSlotted } from "../../utils/dom";
+import {
+  getElementDir,
```
Comment (driskull) [inline src/components/nav-menu-item/nav-menu-item.tsx:17]: `getElementDir` shouldn't be used if at all possible.

File: src/components/nav-menu-item/nav-menu-item.tsx [hunk 2]
```diff
@@ -431,48 +510,52 @@ export class CalciteNavMenuItem {
...
+          class={{
+            container: true,
+            "nav-item-vertical-parent": this.topLevelLayout === "vertical"
+          }}
           role="none"
         >
           <div class="item-content">
             <a
-              aria-current="todo"
+              aria-current="page"
```
Comment (geospatialem) [inline src/components/nav-menu-item/nav-menu-item.tsx:532]: "Current" is currently being read back on all nav menu items, which is great to have for the selected menu item, but not properly identified for nav menu items which are not selected.

Can a conditional be added where the  `aria-current="page"` attribute is read by the currently active page, and other non-selected menu items are set to `aria-current="false"`?

(Similar to this [MDN example](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current#example).)

File: src/components/nav-menu-item/nav-menu-item.tsx [hunk 3]
```diff
@@ -281,6 +330,37 @@ export class CalciteNavMenuItem {
...
+    }
+  };
+
+  private hasSlottedItems(): boolean {
+    return this.el.querySelectorAll("[slot=menu-item-dropdown]").length > 0;
+  }
+
+  private focusFirst(): void {
+    setTimeout(() => this.subMenuItems[0].setFocus(), 1000);
+  }
```
Comment (geospatialem) [inline src/components/nav-menu-item/nav-menu-item.tsx:345]: Think I recall additional R&D is needed on the focus - wanted to point it out again, just to make sure. The delay is a bit disruptive to AT, where if the user tries to press any key, focus is completely lost.

**Review by geospatialem — COMMENTED** [review]
Looking good! A few small behaviors noted regarding a11y, will need to do another manual test when we get further along.

File: src/components/nav-menu-item/nav-menu-item.tsx (hunk 2 above)
Comment (anveshmekala) [reply src/components/nav-menu-item/nav-menu-item.tsx:532]: added an conditional where the `aria-current ="page"` when the nav-menu-item is focused.

**Pull request merged**